phoenix-core/src/it/java/org/apache/phoenix/end2end/CreateTableIT.java

```diff
@@ -801,6 +802,37 @@ public void testSetTableDescriptorPropertyOnView() throws Exception {
         conn2.close();
     }
 
+    @Test
+    public void testCreateViewFromNonExistentTable() throws Exception {
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            conn.createStatement().execute(
+                "CREATE TABLE IF NOT EXISTS S.T1 (A INTEGER PRIMARY KEY, B INTEGER)");
+            // 1. create view from non-existent table (without schema)
+            try {
+                conn.createStatement().execute(
+                    "CREATE VIEW IF NOT EXISTS V1(C INTEGER) AS SELECT * FROM T2");
+            } catch (TableNotFoundException e) {
```

Comment:
can you add a flag that makes sure the code goes to the catch block?
something likes:

flag = false;

try {

} catch {
 flag = true;
}

assertTrue(flag)

Reply:
Done, added `fail()`.
Thanks

Reply:
oh yeah, this is nicer than a flag :)